src/app/pages/jails/jail-list/jail-list.component.ts

```diff
@@ -178,36 +178,54 @@ export class JailListComponent implements OnInit {
     public snackBar: MatSnackBar, public sorter: StorageService, public dialog: MatDialog,) {}
 
 
-  ngOnInit(){
-    this.getActivatedPool();
-    this.getAvailablePools();
-  }
-  afterInit(entityList: any) {
-    this.entityList = entityList;
-  }
-
-  getActivatedPool(){
-    this.ws.call('jail.get_activated_pool').subscribe(
-      (res)=>{
-        if (res != null && res != "") {
-          this.activatedPool = res;
-          this.selectedPool = res;
-          this.isPoolActivated = true;
-        } else {
-          this.isPoolActivated = false;
+  async ngOnInit(){
+    await this.ws.call('pool.query').toPromise().then(
+      (res)=> {
+        this.availablePools = res;
+        if (this.availablePools.length === 0) {
+          this.noPoolDialog();
+          this.toActivatePool = true;
         }
       },
-      (err)=>{
+      (err) => {
         new EntityUtils().handleWSError(this.entityList, err, this.dialogService);
-      })
+      }
+    );
+    if (this.availablePools && this.availablePools.length > 0) {
+      this.ws.call('jail.get_activated_pool').subscribe(
+        (res)=>{
+          if (res != null && res != "") {
+            this.activatedPool = res;
+            this.selectedPool = res;
+            this.isPoolActivated = true;
+          } else {
+            this.isPoolActivated = false;
+          }
+        },
+        (err)=>{
+          new EntityUtils().handleWSError(this.entityList, err, this.dialogService);
+        })
+    }
   }
 
-  getAvailablePools(){
-    this.ws.call('pool.query').subscribe( (res)=> {
-      this.availablePools = res;
+  noPoolDialog() {
+    const dialogRef = this.dialogService.confirm(
+      T('No Pools'),
+      T('Jails cannot be created or managed until a pool is present for storing them.'),
+      true,
+      T('Create Pool'));
```

Comment:
Wait, no cancel button?  What does the user do if they are not ready to create a pool at that moment?

Reply:
It still has a cancel button, it just uses the default which says "Cancel" for the text